src/Common/TransformEndianness.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <base/Decimal_fwd.h>
+#include <base/extended_types.h>
+
+#include <utility>
+
+namespace DB
+{
+template <std::endian endian, typename T>
+    requires is_big_int_v<T>
+inline void transformEndianness(T & x)
+{
+    if constexpr (std::endian::native != endian)
```

Comment:
`std::ranges` is hardly used throughout the codebase and people are not familiar with it (I suspect). It would be nice if you could use good old `std::transform/reverse`, especially since l. 28 does so too.

Reply:
Sure, I don't mind.